app/api/business/brief_overview_business.py

```diff
@@ -1,8 +1,69 @@
 from flask_login import current_user
+
 from app.api.business.agreement_business import use_old_work_order_creator
+from app.api.helpers import abort
+from app.api.services import (audit_service, audit_types,
+                              brief_responses_service, briefs)
+from app.emails import send_opportunity_closed_early_email
+from app.tasks import publish_tasks
+from app.tasks.s3 import create_responses_zip
 from app.validation import get_sections as get_validation_sections
 
 
+def can_close_opportunity_early(brief):
+    seller_selector = brief.data.get('sellerSelector', '')
+    invited_sellers = brief.data.get('sellers', {}).keys()
+    number_of_candidates = int(brief.data.get('numberOfSuppliers', 0))
+
+    if brief.status == 'live' and len(invited_sellers) == 1 and (
+        (brief.lot.slug in ['rfx', 'training2'] and seller_selector == 'oneSeller') or
+        (brief.lot.slug == 'specialist' and seller_selector == 'someSellers')
+    ):
+        invited_seller_code = int(invited_sellers.pop())
+        response_count = brief_responses_service.find(
+            brief_id=brief.id,
+            supplier_code=invited_seller_code,
+            withdrawn_at=None
+        ).count()
```

Comment:
This will need to be adjusted for the fact brief responses now start as drafts and are only submitted when they have a `submitted_at` timestamp value. Right now this code will consider a draft to be valid.

Because I don't think `.find()` can be instructed to match rows that don't have a specific value, you may need to do this instead:

```
response_count = len(
    brief_responses_service.get_brief_responses(brief.id, invited_seller_code, submitted_only=True)
)
```

Reply:
Actually having just typed that out, I believe `BriefResponse` has a computed property of `status`, so maybe you can just add a `status='submitted'` to the `.find()`?

Reply:
I've updated the `.find()` to search for the `submitted` status.